qiskit/circuit/library/standard_gates/pauli.py

```diff
@@ -0,0 +1,88 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Simulator command to perform multiple pauli gates in a single pass
+"""
+import numpy as np
+
+from qiskit.circuit.quantumregister import QuantumRegister
+from qiskit.circuit.library.standard_gates.i import IGate
+from qiskit.circuit.library.standard_gates.x import XGate
+from qiskit.circuit.library.standard_gates.y import YGate
+from qiskit.circuit.library.standard_gates.z import ZGate
+
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.exceptions import CircuitError
+
+
+class PauliGate(Gate):
+    r"""A multi-qubit Pauli gate.
+
+        This gate exists for optimization purposes for the
+        quantum statevector simulation, since applying multiple
+        pauli gates to different qubits at once can be done via
+        a single pass on the statevector.
+
+        The functionality is equivalent to applying
+        the pauli gates sequentially using standard Qiskit gates
+        """
+
+    def __init__(self, pauli_string="I"):
+        if isinstance(pauli_string, float):
+            pauli_string = "I"
+        self.pauli_string = pauli_string
+        super().__init__('pauli', len(pauli_string), [pauli_string[::-1]])
+
+    def _define(self):
+        """
+        gate pauli (p1 a1,...,pn an) { p1 a1; ... ; pn an; }
+        """
+        # pylint: disable=cyclic-import
+        from qiskit.circuit.quantumcircuit import QuantumCircuit
+        gates = {'I': IGate, 'X': XGate, 'Y': YGate, 'Z': ZGate}
+        q = QuantumRegister(len(self.pauli_string), 'q')
+        qc = QuantumCircuit(q, name=self.name)
+
+        rules = [(gates[p](), [q[i]], [])
+                 for (i, p) in enumerate(self.pauli_string)]
```

Comment:
The string format should match the label format for `Pauli` and `Operator` objects, which i think is the reverse of how it is here. eg `Operator.from_label('XI')` is `QuantumCircuit(2).x(1)`

I think this would be fixed by
```suggestion
                 for (i, p) in enumerate(reversed(self.pauli_string))]
```

Reply:
It seems better to reverse the Pauli string once, in line 43 (also, I have a bad history with `reversed` and prefer to use `[::-1]`)